### PR TITLE
Breaking: Bump to typescript-estree@6.0.0, typescript@~3.2.1

### PR DESCRIPTION
**Pull request by nevir**
From what I can tell, the main change in typescript-estree is added support for `BigIntLiteral`s. Added some snapshots for 'em, as well.

---

Should this be a major bump due to the typescript upgrade? Or only minor?

### CODE REVIEW

**Comment by armano2** [comment]
can i add new nodes to `visitor-keys`

**Comment by kaicataldo** [comment]
Given the discussion in [this PR](https://github.com/eslint/typescript-eslint-parser/pull/573), the team seems to be in general consensus that the "supported TS version" change is a breaking change.

**Comment by nevir** [comment]
> can you add new nodes to `visitor-keys`

Hmm, I'm not quite sure which nodes would need to be added to `visitor-keys` (and by that, you mean [./visitor-keys.js](https://github.com/eslint/typescript-eslint-parser/blob/master/visitor-keys.js)?). Seems like this is only adding a new literal value type ([`BigIntLiteral `](https://github.com/eslint/typescript-eslint-parser/pull/576/files#diff-8f3e08f2fe2cde310f4f60301a9daae7R9165)), and that doesn't seem to fit with the structure of that file. I admit, though, I'm not too familiar w/ how eslint juggles the AST.

> Do you mind also updating the readme with the supported TS version range?

👍  Done.  Also switched the version constraint to a `^`, as suggested in https://github.com/eslint/typescript-eslint-parser/pull/573#issuecomment-443797291

**Comment by armano2** [comment]
@nevir you should just add `"BigIntLiteral": []` to ./visitor-keys.js

**Comment by platinumazure** [comment]
Is `BigIntLiteral` a node type? Or is it a type of the existing `Literal` node?

**Comment by armano2** [comment]
its new node type

**Comment by justinanastos** [comment]
I just wanted to chime in and say thank you to everyone who's working on this! Great work!

**Comment by nevir** [comment]
Added the node type

**Comment by nevir** [comment]
@kaicataldo, is there anything else you'd like me to tweak?

File: package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
   "dependencies": {
     "eslint-scope": "^4.0.0",
     "eslint-visitor-keys": "^1.0.0",
-    "typescript-estree": "5.3.0"
+    "typescript-estree": "^6.0.0"
```
Comment (KaelWD) [inline package.json]: 7.0.0 just came out

Comment (armano2) [reply package.json]: @KaelWD i did pr with changes for 7.0.0 in #584

**Comment by platinumazure** [comment]
Hi @nevir, apologies for letting this sit over the holidays.

Is this PR still needed now that #589 has been created?

**Comment by JamesHenry** [comment]
Thanks a lot for this @nevir and I am sorry for the delay - your commits (with you still as the author) are included directly in https://github.com/eslint/typescript-eslint-parser/pull/596, so I am closing this in favour of that one.

Thanks again!

**Pull request closed without merging**